### PR TITLE
Add a few functions & POSIX error codes to Windows API

### PR DESCRIPTION
**Pull request by mikehoyle**
This change adds a few simple additional functions to the windows apis, as well as some additional error codes.

Tests and style pass, as per contributing guidelines.

### CODE REVIEW

**Comment by retep998** [comment]
`ucrt\errno.h` does have those error constants, and notably they are *different* than the WSA constants of the same name. For example `EADDRINUSE` from ucrt has a value of 100 but `EADDRINUSE` from winsock2 has a value of 10048. Obviously a massive footgun but it is what it is.

`time` and `_lseeki64` are both from ucrt as well. Notably however `time` is actually an inline wrapper around `_time64`. The signatures look correct otherwise.

**Comment by UdamSaini** [comment]
Inside winsock.h, they say to not use EADDRINUSE and to use WSAEADDRINUSE due to the conflict for socket related errors. 
/*
* Windows Sockets errors redefined as regular Berkeley error constants.
* These are commented out in Windows NT to avoid conflicts with errno.h.
* Use the WSA constants instead.
*/
#if 0
#define EWOULDBLOCK  WSAEWOULDBLOCK
......

I think we should probably use the errno constants as that is what windows uses, and have the WSA constants be separate or in the win32 library? I wonder if we should add a comment to that effect, similar to the windows comment.

**Comment by mikehoyle** [comment]
In my most recent commit I updated the time funtion to link to _time64 (as time is just a wrapper around that anyways). It also seems that the gnu toolchain uses a different (probably outdated) version of errno.h that doesn't contain EOTHER, so I have moved that specific constant to the msvc-specific file.

File: src/windows/mod.rs [hunk 1]
```diff
@@ -183,6 +183,48 @@ pub const ENOTEMPTY: ::c_int = 41;
 pub const EILSEQ: ::c_int = 42;
 pub const STRUNCATE: ::c_int = 80;
 
+// POSIX Supplement (from errno.h)
+pub const EADDRINUSE: ::c_int = 100;
```
Comment (gnzlbg) [inline src/windows/mod.rs]: Per the comment below from @UdamSaini , does it make sense to add this one here ? (or maybe we can add it as `#[deprecated]` with the same warning as the windows headers?

Comment (retep998) [reply src/windows/mod.rs]: Those warnings are on the winsock definitions that just redefine to the WSA constants, not the errno definitions.

**Pull request merged**